crates/biome_js_analyze/src/analyzers/nursery/no_default_export.rs

```diff
@@ -0,0 +1,82 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::AnyJsExportClause;
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow `export default`.
+    ///
+    /// In contrast to default exports,
+    /// named exports are discoverable in an Editor: the editor can suggest to import a named export.
+    /// Also, named exports improve consistency by using the same name through a code base.
+    ///
+    /// Moreover, default exports encourage exporting an object that acts as a namespace.
+    /// This is a legacy pattern used to mimic CommonJS modules.
+    ///
+    /// Source: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-default-export.md
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// export default function f() {};
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// export default class C {};
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// export default {
+    ///     f() {},
+    ///     g() {},
+    /// };
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// export function f () {};
+    /// export class C {};
+    /// ```
+    ///
+    pub(crate) NoDefaultExport {
+        version: "next",
+        name: "noDefaultExport",
+        recommended: false,
+    }
+}
+
+impl Rule for NoDefaultExport {
+    type Query = Ast<AnyJsExportClause>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let export_clause = ctx.query();
+        matches!(
+            export_clause,
+            AnyJsExportClause::JsExportDefaultDeclarationClause(_)
+                | AnyJsExportClause::JsExportDefaultExpressionClause(_)
+        )
+        .then_some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let export_clause = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                export_clause.range(),
+                markup! {
+                    "Use a named export instead."
```

Comment:
Based on [our pillars](https://biomejs.dev/linter/#rule-pillars), we should fix and explain the error, so this message should tell that a default export was detected and it's an error.

Then, we should explain why it's an error, maybe by telling the users about discoverability. Then, at the end, we can suggest using a named export.